### PR TITLE
feat: add invisible variant for drawer

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #2478

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/styles/_dialog-init.scss [hunk 1]
```diff
@@ -1,20 +1,10 @@
 @use "@db-ui/foundations/build/scss/colors";
 
 $backdrop-color: var(
-	--db-current-color-enabled,
-	var(--db-neutral-on-bg-enabled, #242629)
+	#{colors.$db-current-color-enabled},
+	var(#{colors.$db-neutral-on-bg-enabled}, #242629)
```
Comment (bruno-sch) [inline packages/components/src/styles/_dialog-init.scss:5]: Maybe you can explain it to me: What is the idea behind using a HEX value as an "unsafe" fallback?

File: packages/components/src/styles/_dialog-init.scss [hunk 2]
```diff
@@ -34,13 +24,20 @@ dialog {
 
 	&:not([data-backdrop="none"]) {
 		&::backdrop {
-			@extend %backdrop;
+			background-color: $backdrop-color;
+			opacity: 0.64;
```
Comment (bruno-sch) [inline packages/components/src/styles/_dialog-init.scss:28]: The value for opacity is repeated at various points. Wouldn't it make sense to use global variables for this?

File: showcases/patternhub/styles/globals.scss [hunk 3]
```diff
@@ -397,7 +397,8 @@ h6 {
 
 		.backdrop {
 			border-radius: variables.$db-border-radius-sm;
-			@extend %backdrop;
+			background-color: dialog-init.$backdrop-color;
+			opacity: 0.64;
```
Comment (bruno-sch) [inline showcases/patternhub/styles/globals.scss:401]: The value for opacity is repeated at various points. Wouldn't it make sense to use global variables for this?

File: packages/components/src/styles/_dialog-init.scss (hunk 1 above)
Comment (nmerget) [reply packages/components/src/styles/_dialog-init.scss]: removed it, was a leftover of an old implementation

File: packages/components/src/styles/_dialog-init.scss (hunk 2 above)
Comment (nmerget) [reply packages/components/src/styles/_dialog-init.scss]: fixed

File: showcases/patternhub/styles/globals.scss (hunk 3 above)
Comment (nmerget) [reply showcases/patternhub/styles/globals.scss]: fixed

**Review by bruno-sch — APPROVED** [review]

File: packages/components/src/styles/_dialog-init.scss [hunk 4]
```diff
@@ -34,13 +27,20 @@ dialog {
...
+			opacity: $backdrop-opacity-strong;
 		}
-	}
 
-	&[data-backdrop="weak"] {
-		&::backdrop {
-			@extend %backdrop-weak;
+		&[data-backdrop="invisible"] {
+			&::backdrop {
+				opacity: 0;
```
Comment (mfranzke) [inline packages/components/src/styles/_dialog-init.scss:36]: doesn't this totally hide it ? What's the difference in between `dialog.showModal()` against `dialog.show()` in this case anyhow ?

**Pull request merged**